packages/epo-widget-lib/src/widgets/ColorTool/ColorTool.tsx

```diff
@@ -0,0 +1,203 @@
+import { FormEvent, FunctionComponent, useState, useEffect } from "react";
+import useResizeObserver from "use-resize-observer";
+import { useTranslation } from "react-i18next";
+import {
+  getCategoryName,
+  getDataFiltersByName,
+  isResetButtonActive,
+  prepareData,
+  resetFilters,
+} from "./utilities";
+import * as Styled from "./styles";
+import { Option } from "@rubin-epo/epo-react-lib/dist/types/select";
+import { ListboxOption } from "@rubin-epo/epo-react-lib/dist/types/select-listbox";
+import { Select } from "@rubin-epo/epo-react-lib";
+import FilterControls from "./FilterControls";
+import ImageComposite from "./ImageComposite";
+
+export interface ImageFilter {
+  label: string;
+  defaultColor?: string;
+  color?: string;
+  active: boolean;
+  image: string;
+  isDisabled: boolean;
+  value: number;
+  defaultValue?: number;
+  min: number;
+  max: number;
+  brightness?: number;
+}
+
+export interface AstroObject {
+  name: string;
+  caption?: string;
+  filters: ImageFilter[];
+}
+
+export interface AstroCategory {
+  type: string;
+  objects: AstroObject[];
+}
+
+interface ColorToolProps {
+  selectionCallback: (selectedData: AstroObject) => void;
+  data: AstroCategory[];
+  selectedData: AstroObject;
+  title?: string;
+  objectOptions?: Option[];
+  colorOptions?: ListboxOption[];
+  isDisabled?: boolean;
+  hideControls?: boolean;
+  hideImage?: boolean;
+  hideSubtitle?: boolean;
+}
+
+const ColorTool: FunctionComponent<ColorToolProps> = ({
+  title,
+  data,
+  objectOptions = [],
+  selectedData: preSelectedData,
+  colorOptions = [],
+  selectionCallback,
+  isDisabled = false,
+  hideControls = false,
+  hideImage = false,
+  hideSubtitle = false,
+}) => {
+  const { ref, width = 1 } = useResizeObserver<HTMLDivElement>();
```

Comment:
`useResizeObserver` is used to get the width of the widget so we can determine if it's condensed/full size since container queries don't work yet (😢 )